libraries/AP_Scripting/generator/src/main.c

```diff
@@ -1386,41 +1386,41 @@ void emit_checker(const struct type t, int arg_number, int skipped, const char *
     arg_number = arg_number + NULLABLE_ARG_COUNT_BASE;
     switch (t.type) {
       case TYPE_BOOLEAN:
-        fprintf(source, "%sbool data_%d;\n", indentation, arg_number);
+        fprintf(source, "%sbool data_%d {};\n", indentation, arg_number);
```

Comment:
isn't the right fix to add checksum=0; in AP_Filesystem.cpp?
changing the scripting API would seem to cost a lot more flash, and this PR doesn't fix the real bug which is in AP_Filesystem and could impact non-scripting users

Reply:
None of the current users do it, but without zeroing means you can set the checksum initial value. Either to something in particular or you could crc several files in turn.

I'm happy to fix the file system too, but my real concern is that we have other bindings that have the same issue and no one has spotted it.

Reply:
I have moved this change to a https://github.com/ArduPilot/ardupilot/pull/26222 and added the file system fix here.